### PR TITLE
fix(model-server): writing to a branch using the v1 API failed

### PR DESCRIPTION
**Pull request by slisson**
Ignite doesn't support nested transactions. Starting a transaction from inside a transaction throws the following exception:

```
java.lang.IllegalStateException: Failed to start new transaction (current thread already has a transaction): ...
	at org.apache.ignite.internal.processors.cache.transactions.IgniteTransactionsImpl.txStart0(IgniteTransactionsImpl.java:184)
	at org.apache.ignite.internal.processors.cache.transactions.IgniteTransactionsImpl.txStart(IgniteTransactionsImpl.java:70)
	at org.modelix.model.server.store.IgniteStoreClient.runTransaction(IgniteStoreClient.kt:139)
	at org.modelix.model.server.handlers.RepositoriesManager.mergeChanges(RepositoriesManager.kt:151)
	at org.modelix.model.server.handlers.KeyValueLikeModelServer$putEntries$1.invoke(KeyValueLikeModelServer.kt:323)
	at org.modelix.model.server.handlers.KeyValueLikeModelServer$putEntries$1.invoke(KeyValueLikeModelServer.kt:316)
	at org.modelix.model.server.store.IgniteStoreClient.runTransaction(IgniteStoreClient.kt:140)
	at org.modelix.model.server.handlers.KeyValueLikeModelServer.putEntries(KeyValueLikeModelServer.kt:316)
	at org.modelix.model.server.handlers.KeyValueLikeModelServer$modelServerModule$1$3$7.invokeSuspend(KeyValueLikeModelServer.kt:163)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.ktor.server.netty.EventLoopGroupProxy$Companion.create$lambda$1$lambda$0(NettyApplicationEngine.kt:296)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```

### CODE REVIEW

**Review by odzhychko — APPROVED** [review]

File: commitlint.config.js [hunk 1]
```diff
@@ -28,4 +28,7 @@ module.exports = {
     ],
     "subject-case": [0, 'never']
   },
+  ignores: [
+    (message) => message.includes('skip-lint')
+  ],
```
Comment (languitar) [inline commitlint.config.js:33]: Can we lift this change into a separate chore/build commit?

**Review by languitar — CHANGES_REQUESTED** [review]
A test verifying this bug fix would be a good addition here to ensure this remains working.

**Comment by slisson** [comment]
The new tests revealed unknown bugs. Looks like I have to fix them as well.

**Comment by slisson** [comment]
The failing new tests will be fixed in a separate PR https://github.com/modelix/modelix.core/pull/343, because they are unrelated to the transaction handling bug fixed by this PR.

**Comment by languitar** [comment]
As the test is not part of the commit that fixes the bug, I am lost which of tests is actually intended to verify the fixed bug. Can you give me a pointer?

**Comment by slisson** [comment]
There are now unit tests that ensure the Ignite and map based implementation of IStoreClient behave the same.

**Pull request merged**